backend/src/main/java/org/eclipse/tractusx/semantics/registry/repository/ShellRepository.java

```diff
@@ -53,13 +53,11 @@ public interface ShellRepository extends PagingAndSortingRepository<Shell, UUID>
      * @return external shell ids for the given key value combinations
      */
     @Query(
-            "select s.id_external from shell s where s.id in (" +
-                    "select si.fk_shell_id from shell_identifier si " +
+            "select s.id_external from shell s join shell_identifier si on s.id = si.fk_shell_id " +
                     "where concat(si.namespace,si.identifier) in (:keyValueCombinations) " +
-                    "AND ((si.external_subject_id is not null and si.external_subject_id = :tenantId) or :tenantId = s.tenant_id) " +
-                    "group by si.fk_shell_id " +
-                    "having count(*) = :keyValueCombinationsSize " +
-            ")"
+                    "AND ((si.external_subject_id is not null and si.external_subject_id = :tenantId) " +
```

Comment:
is this redundant condition - `si.external_subject_id is not null ` . Can it work with this condition alone -  `si.external_subject_id = :tenantId` ?

Reply:
This is an old query only and we have removed the IN condition and replaced it with JOIN.

Reply:
Thanks for the clarification @rahulrkhapre . The performance will improve if we remove redundant condition.